File: tests/agent_workloads/adapters/desktop_interaction_focus.mjs
import { readFileSync } from "node:fs";

if (process.argv.length !== 3) {
  throw new Error("usage: desktop_interaction_focus INPUT_FILE");
}

const focus = ["Save", "Cancel"];
let index = 1;
for (const [lineNumber, key] of readFileSync(process.argv[2], "utf8")
  .split(/\r?\n/)
  .entries()) {
  if (lineNumber === 0 || !key) continue;
  if (key === "Tab") {
    console.log(`focus|${focus[index]}`);
    index = (index + 1) % focus.length;
  } else if (key === "Empty") {
    console.log("event|Empty|observed");
  } else {
    console.log(`event|${key}|observed`);
  }
}

File: tests/agent_workloads/adapters/desktop_interaction_focus.py
import sys
from pathlib import Path


if len(sys.argv) != 2:
    raise SystemExit("usage: desktop_interaction_focus INPUT_FILE")

focus = ["Save", "Cancel"]
index = 1
for line_number, key in enumerate(Path(sys.argv[1]).read_text().splitlines(), 1):
    if line_number == 1 or not key:
        continue
    if key == "Tab":
        print(f"focus|{focus[index]}")
        index = (index + 1) % len(focus)
    elif key == "Empty":
        print("event|Empty|observed")
    else:
        print(f"event|{key}|observed")
